packages/next/client/head-manager.ts

```diff
@@ -119,14 +121,21 @@ export default function initHeadManager(): {
   updateHead: (head: JSX.Element[]) => void
 } {
   let updatePromise: Promise<void> | null = null
+  function promiseTransition(fn: Function) {
+    const promise = (updatePromise = Promise.resolve().then(() => {
+      if (promise !== updatePromise) return
+      updatePromise = null
+      fn()
+    }))
+  }
+  const startTransition = process.env.__NEXT_REACT_ROOT
+    ? (React as any).startTransition
+    : promiseTransition
 
   return {
     mountedInstances: new Set(),
     updateHead: (head: JSX.Element[]) => {
-      const promise = (updatePromise = Promise.resolve().then(() => {
-        if (promise !== updatePromise) return
-
-        updatePromise = null
+      startTransition(() => {
```

Comment:
> Route announcer effects are executed later than `document.title` updating in react 17 but earlier than it in react 18.
> 
> The `updateHead` method now is using promise based transition in react 17, and opt-in `startTransition` in react 18

@huozhi 

If I understand it correctly, the issue is that the React 18 update the head **after** `setRouteAnnoucer` effect is committed, causing the `setRouteAnnoucer` is using the not up-to-dated `document.title`. But why are you trying to delay the head updating with `startTransition` though? I thought the `setRouteAnnoucer` should be delayed instead.

Reply:
That doesn't work, because in that way the document.title update is still after the head manager updating the title.

```
head effects ------------> update document title
route annoucer effects
```

So startTransition won't work in that case.